src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/wa/CftQueryServiceITTest.java

```diff
@@ -565,6 +566,61 @@ void should_retrieve_tasks_when_role_assignment_specific_and_excluded() {
 
     }
 
+    @Test
+    void should_retrieve_tasks_when_task_type_filter_applied() {
+
+        List<String> caseIds = List.of("1623278362400035", "1623278362400036");
+        String taskType = "followUpOverdueCaseBuilding";
+        SearchTaskRequest searchTaskRequest = new SearchTaskRequest(List.of(
+            new SearchParameterList(TASK_TYPE, SearchOperator.IN, singletonList(taskType)),
+            new SearchParameterList(CASE_ID, SearchOperator.IN, caseIds)
+        ));
+
+        List<RoleAssignment> roleAssignments = new ArrayList<>();
+
+        //apply standard role to user
+        RoleAssignmentRequest roleAssignmentRequest = RoleAssignmentRequest.builder()
+            .testRolesWithGrantType(TestRolesWithGrantType.STANDARD_SENIOR_TRIBUNAL_CASE_WORKER_PUBLIC)
+            .roleAssignmentAttribute(
+                RoleAssignmentAttribute.builder()
+                    .jurisdiction(WA_JURISDICTION)
+                    .region("1")
+                    .baseLocation(PRIMARY_LOCATION)
+                    .build()
+            )
+            .build();
+
+        createRoleAssignment(roleAssignments, roleAssignmentRequest);
+
+        AccessControlResponse accessControlResponse = new AccessControlResponse(userInfo, roleAssignments);
+        permissionsRequired.add(PermissionTypes.READ);
+
+        GetTasksResponse<Task> allTasks = cftQueryService.searchForTasks(
+            0,
+            10,
+            searchTaskRequest,
+            accessControlResponse,
+            false
+        );
+
+
+        //standard user can retrieve task
+        Assertions.assertThat(allTasks.getTotalRecords())
+            .isEqualTo(1);
+
+        Assertions.assertThat(allTasks.getTasks())
+            .hasSize(1)
+            .flatExtracting(Task::getCaseId)
+            .containsExactly(caseIds.get(0));
+
+        //allTasks.getTasks().get(0).getType()
+        Assertions.assertThat(allTasks.getTasks())
+            .hasSize(1)
+            .flatExtracting(Task::getType)
+            .containsExactly(taskType);
+
+    }
+
```

Comment:
Can we introduce a test that searches by more than one taskType and returns at least 2 tasks for the different taskTypes?

Reply:
new scenario added